lib/chef-dk/command/generator_commands/cookbook.rb

```diff
@@ -193,10 +202,7 @@ def contains_delivery_cli_toml?(path)
         def read_and_validate_params
           arguments = parse_options(params)
           @cookbook_name_or_path = arguments[0]
-          if !@cookbook_name_or_path
-            @params_valid = false
-          elsif /-/ =~ File.basename(@cookbook_name_or_path)
-            err("Hyphens are not allowed in cookbook names. Please specify a cookbook name without hyphens.")
+          unless @cookbook_name_or_path
```

Comment:
`@cookbook_name_or_path` is accessed elsewhere. It also makes it a little easier to read when referenced in my opinion.